txpipe/extensions/twopoint_scia.py

```diff
@@ -0,0 +1,493 @@
+            else:
+                rancat = None
+        else:
+            if self.config['var_method']=='jackknife':
+                patch_centers = self.get_input('patch_centers')
+                cat = treecorr.Catalog(
+                    ra=ra, dec=dec,
+                    ra_units='degree', dec_units='degree',
+                    patch_centers=patch_centers)
+            else:
+                cat = treecorr.Catalog(
+                    ra=ra, dec=dec,
+                    ra_units='degree', dec_units='degree')
+
+            if 'random_bin' in data:
+                random_mask = data['random_bin']==i
+                if self.config['var_method']=='jackknife':
+                    rancat  = treecorr.Catalog(
+                        ra=data['random_ra'][random_mask], dec=data['random_dec'][random_mask],
+                        ra_units='degree', dec_units='degree',
+                        patch_centers=patch_centers)
+                else:
+                    rancat  = treecorr.Catalog(
+                        ra=data['random_ra'][random_mask], dec=data['random_dec'][random_mask],
+                        ra_units='degree', dec_units='degree')
+            else:
+                rancat = None
+
+        return cat, rancat
+
+    def get_shear_catalog(self, data, meta, i):
+        import treecorr
+        import pyccl as ccl
+        g1, g2, mask = self.get_calibrated_catalog_bin(data, meta, i)
+        
+        if self.config['3Dcoords']:
+            mu = data['mu'][mask]
+            cosmo = ccl.Cosmology.read_yaml(self.get_input('fiducial_cosmology'))
+            r = ccl.background.comoving_radial_distance(cosmo, 1/(1+mu))
+
+            if self.config['var_method']=='jackknife':
+                patch_centers = self.get_input('patch_centers')
+                cat = treecorr.Catalog(
+                    g1 = g1,
+                    g2 = g2,
+                    r = r,
+                    ra = data['ra'][mask],
+                    dec = data['dec'][mask],
+                    ra_units='degree', dec_units='degree',
+                    patch_centers=patch_centers)
+                    #npatch=self.config['npatch'])
+            else:
+                cat = treecorr.Catalog(
+                    g1 = g1,
+                    g2 = g2,
+                    r = r,
+                    ra = data['ra'][mask],
+                    dec = data['dec'][mask],
+                    ra_units='degree', dec_units='degree')
+        else:
+            if self.config['var_method']=='jackknife':
+                patch_centers = self.get_input('patch_centers')
+                cat = treecorr.Catalog(
+                    g1 = g1,
+                    g2 = g2,
+                    ra = data['ra'][mask],
+                    dec = data['dec'][mask],
+                    ra_units='degree', dec_units='degree',
+                    patch_centers=patch_centers)
+                    #npatch=self.config['npatch'])
```

Comment:
not sure what was this for

Reply:
Old attempt at the code, before we decided on building the jackknife centers before hand. cleaned it up!